src/main/java/org/jenkinsci/plugins/enhancedoldbuilddiscarder/EnhancedOldBuildDiscarder.java

```diff
@@ -211,6 +209,9 @@ public void perform(Job<?,?> job) throws IOException, InterruptedException {
 			int newCntr = 0;
 			Calendar cal = new GregorianCalendar();
 			cal.add(Calendar.DAY_OF_YEAR,-daysToKeep);
+			if (ageUnitTest()){ // used to test the age conditions for the hold max builds feature
+				cal.add(Calendar.DAY_OF_YEAR,2*(daysToKeep));
+			}
```

Comment:
Maybe instead of trying to detect if its a test, just make a new method that returns the cal/days to keep?

private Calendar getCalDaysToKeep(int daysToKeep) {
  Calendar cal = new GregorianCalendar();
  cal.add(Calendar.DAY_OF_YEAR,-daysToKeep);
  return cal
}

Then your test can just override that function

Reply:
I prefer this as well, I have implemented these suggestions :)